### ISSUE TITLE
Refactor ECPoint for NEO3x

### ISSUE THREAD

**Issue by vncoelho** [neo-project/neo#1654]
`ECPoint` is used in different parts of the code, however, in principle, in most of its use cases we could direct use `pubkey` as `byte[]` itself.

**Comment by vncoelho** [comment]
Maybe now with support to more cryptography standards there will be more use cases https://github.com/neo-project/neo/commit/1af936929093721ec274167829761ddba825c86c

**Comment by vncoelho** [comment]
However, for a more generic use the function `public static ECPoint DecodePoint(ReadOnlySpan<byte> encoded, ECCurve curve)` should use the `This.Curve` of the object.

**Comment by doubiliu** [comment]
Agree, this can better encapsulate the parsing process of publickey in functions such as signature verification, which is conducive to supporting more encryption standards.This is consistent with the principle of cohesion

**Comment by shargon** [comment]
> However, for a more generic use the function `public static ECPoint DecodePoint(ReadOnlySpan<byte> encoded, ECCurve curve)` should use the `This.Curve` of the object.

is the expected curve, so there is no `this` already
